### ISSUE TITLE
Error: You need to run with a version of node that supports ES Modules in the VM API

### ISSUE THREAD

**Issue by seeren** [jmcdo29/nest-commander#1077]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Potential Commit/PR that introduced the regression

65f5abce1e04528fa8a70b2a351c904e96db1fc2

### Versions

3.12.2 -> 3.12.3

### Describe the regression

Jest build fail when import `'nest-commander'` members with following error:

```bash
 RUNS  src/app/core/questions/confirm.question.spec.ts
/apps/my-project/node_modules/jest-util/build/invariant.js:16
    throw new Error(message);
          ^
Error: You need to run with a version of node that supports ES Modules in the VM API. See https://jestjs.io/docs/ecmascript-modules
    at invariant (/apps/aftersales-api/node_modules/jest-util/build/invariant.js:16:11)
    at importModuleDynamically (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1511:37)
    at importModuleDynamicallyWrapper (node:internal/vm/module:428:21)
    at importModuleDynamicallyCallback (node:internal/modules/esm/utils:85:14)
    at Object.<anonymous> (/apps/aftersales-api/node_modules/nest-commander/node_modules/prettier/index.cjs:593:23)
    at Runtime._execModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1439:24)
    at Runtime._loadModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1022:12)
    at Runtime.requireModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:882:12)
    at Runtime.requireModuleOrMock (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1048:21)
    at Object.<anonymous> (/apps/aftersales-api/node_modules/nest-commander/node_modules/@fig/complete-commander/lib/index.js:16:36)
    at Runtime._execModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1439:24)
    at Runtime._loadModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1022:12)
    at Runtime.requireModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:882:12)
    at Runtime.requireModuleOrMock (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1048:21)
    at Object.<anonymous> (/apps/aftersales-api/packages/nest-commander/src/completion.factory.ts:2:1)
    at Runtime._execModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1439:24)
    at Runtime._loadModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1022:12)
    at Runtime.requireModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:882:12)
    at Runtime.requireModuleOrMock (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1048:21)
    at Object.<anonymous> (/apps/aftersales-api/packages/nest-commander/src/command.factory.ts:14:1)
    at Runtime._execModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1439:24)
    at Runtime._loadModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1022:12)
    at Runtime.requireModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:882:12)
    at Runtime.requireModuleOrMock (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1048:21)
    at Object.<anonymous> (/apps/aftersales-api/packages/nest-commander/src/index.ts:3:1)
    at Runtime._execModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1439:24)
    at Runtime._loadModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1022:12)
    at Runtime.requireModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:882:12)
    at Runtime.requireModuleOrMock (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1048:21)
    at Object.<anonymous> (/apps/aftersales-api/src/app/core/questions/confirm.question.ts:13:26)
    at Runtime._execModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1439:24)
    at Runtime._loadModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1022:12)
    at Runtime.requireModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:882:12)
    at Runtime.requireModuleOrMock (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1048:21)
    at Object.<anonymous> (/apps/aftersales-api/src/app/core/questions/confirm.question.spec.ts:4:28)
    at Runtime._execModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1439:24)
    at Runtime._loadModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:1022:12)
    at Runtime.requireModule (/apps/aftersales-api/node_modules/jest-runtime/build/index.js:882:12)
    at jestAdapter (/apps/aftersales-api/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:77:13)
    at runTestInternal (/apps/aftersales-api/node_modules/jest-runner/build/runTest.js:367:16)
    at runTest (/apps/aftersales-api/node_modules/jest-runner/build/runTest.js:444:34)

```

### Minimum reproduction code

Every source code that import `'nest-commander'` members, a simple example:

```ts
import { Question, QuestionSet } from 'nest-commander';

@QuestionSet({ name: 'confirm' })
export class ConfirmQuestion {
  @Question({
    type: 'confirm',
    name: 'confirm',
    message: 'Do you confirm?',
    default: true,
  })
  parseConfirm(val: boolean) {
    return val;
  }
}
```

Then the error occurs when running tests with jest:

`jest --verbose --config ./jest-spec.json`

Using configuration file:

```json
{
  "moduleFileExtensions": ["js", "json", "ts"],
  "rootDir": ".",
  "modulePaths": ["./"],
  "testRegex": ".*\\.spec\\.ts$",
  "transform": {
    "^.+\\.(t|j)s$": "ts-jest"
  },
  "clearMocks": true,
  "testEnvironment": "node"
}

```

Test example:

```ts
import { Test, TestingModule } from '@nestjs/testing';

import { ConfirmQuestion } from 'src/app/core/questions/confirm.question';

describe('ConfirmQuestion', () => {
  let module: TestingModule;
  let confirmQuestion: ConfirmQuestion;

  beforeAll(async (): Promise<TestingModule> => {
    module = await Test.createTestingModule({
      providers: [ConfirmQuestion],
    }).compile();
    confirmQuestion = module.get(ConfirmQuestion);
    return module;
  });

  afterAll(async (): Promise<void> => module.close());

  describe('parseConfirm', () => {
    it('Retrieve argument', (): void => {
      expect(confirmQuestion.parseConfirm(true)).toBeTruthy();
    });
  });
});
```

### Expected behavior

I expect no error occurs when running tests with jest

**Comment by jmcdo29** [comment]
Can you make a repo I can clone to test with?

**Comment by mareksuscak** [comment]
![CleanShot 2023-12-02 at 01 37 58@2x](https://github.com/jmcdo29/nest-commander/assets/613647/ef3ab769-1b13-4ebe-bd65-de4e9e9e616d)

We're seeing a similar error here. Also happens when a command gets imported into a unit test with Jest.

**Comment by jmcdo29** [comment]
Please provide a [minimum reproduction repository](https://minimum-reproduction.wtf/) (Git repository/[StackBlitz](https://stackblitz.com/fork/github/nestjs/typescript-starter)/[CodeSandbox](https://codesandbox.io/s/github/nestjs/typescript-starter/tree/master) project).

## [why reproductions are required](https://antfu.me/posts/why-reproductions-are-required)

**Comment by jmcdo29** [comment]
Thank you all for the reproductions, very helpful in testing out a fix locally.

I think for now, I'm going to move the `@fig/commander-completion` to an optional import and throw an error if it somehow doesn't get `require`d correctly. From a cursory test, this should fix up the errors from Jest.

**Comment by jmcdo29** [comment]
Should be resolved by 3.12.5
